From 14dbd16e434fe05010e4c3a0b6941378b64383b2 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 15 Jun 2022 12:25:19 +0300
Subject: [PATCH] [SW-0] gaudi3: remove sram bmu configuration

Since ROM doesn't support secured firmware update as part of recovery,
this patch removes SRAM BMU configuration performed as part of
PCIe link configuration routine of ROM recovery. In ROM recovery all
config space is opened based on efuse value and firmware loader
can perform non-secured firmware update.

Change-Id: I66b3e4f62d933a81e5ea126ca4d8f95dfd3d1244
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index b6d614a..f4b2095 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -216,17 +216,9 @@
 	regmap_writel(BMU_CFG_0_DST_OFFSET_R0, mmD0_PCIE_BMU_CFG_DST_OFFS_R0_0);
 	regmap_writel(BMU_CFG_0_DST_OFFSET_R1, mmD0_PCIE_BMU_CFG_DST_OFFS_R1_0);
 
-	/* Region 0 MEM region SRAM BAR 4/5 */
-	/* upper 16 bits of 0x0200_FFFF_F000_0000 */
-	regmap_writel(BMU_MEM_0_DST_BASE, mmD0_PCIE_BMU_MEM_DST_BASE_0);
-	regmap_writel(BMU_MEM_0_SRC_START, mmD0_PCIE_BMU_MEM_SRC_START_0);
-	regmap_writel(BMU_MEM_0_SRC_END, mmD0_PCIE_BMU_MEM_SRC_END_0);
-	/* bits[47:20] of 0x0200_FFFF_F000_0000 */
-	regmap_writel(BMU_MEM_0_DST_OFFSET, mmD0_PCIE_BMU_MEM_DST_OFFS_0);
-
-	/* Enable Region 0 CFG and MEM and Region 1 MEM */
+	/* Enable Region 0 CFG */
 	val = regmap_readl(mmD0_PCIE_BMU_CFG_EN);
-	val |= D0_PCIE_BMU_CFG_EN_F0_CFG_R0_M | D0_PCIE_BMU_CFG_EN_F0_MEM_R0_M;
+	val |= D0_PCIE_BMU_CFG_EN_F0_CFG_R0_M;
 	regmap_writel(val, mmD0_PCIE_BMU_CFG_EN);
 
 	/* Enable BMU */
